gluten-core/src/main/scala/io/glutenproject/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -469,52 +468,31 @@ object FilterHandler {
 
   // Separate and compare the filter conditions in Scan and Filter.
   // Push down the left conditions in Filter into Scan.
-  def applyFilterPushdownToScan(plan: FilterExec, reuseSubquery: Boolean): SparkPlan =
-    plan.child match {
+  def applyFilterPushdownToScan(filter: FilterExec, reuseSubquery: Boolean): SparkPlan =
+    filter.child match {
       case fileSourceScan: FileSourceScanExec =>
         val leftFilters =
-          getLeftFilters(fileSourceScan.dataFilters, flattenCondition(plan.condition))
-        // transform BroadcastExchangeExec to ColumnarBroadcastExchangeExec in partitionFilters
-        val newPartitionFilters =
-          ExpressionConverter.transformDynamicPruningExpr(
-            fileSourceScan.partitionFilters,
-            reuseSubquery)
-        new FileSourceScanExecTransformer(
-          fileSourceScan.relation,
-          fileSourceScan.output,
-          fileSourceScan.requiredSchema,
-          newPartitionFilters,
-          fileSourceScan.optionalBucketSet,
-          fileSourceScan.optionalNumCoalescedBuckets,
-          fileSourceScan.dataFilters ++ leftFilters,
-          fileSourceScan.tableIdentifier,
-          fileSourceScan.disableBucketedScan
-        )
+          getLeftFilters(fileSourceScan.dataFilters, flattenCondition(filter.condition))
+        ScanTransformerFactory.createFileSourceScanTransformer(
+          fileSourceScan,
+          reuseSubquery,
+          extraFilters = leftFilters)
       case batchScan: BatchScanExec =>
```

Comment:
Thanks for your fix. Can be avoid the extra filter pushdown for BatchScan by not calling `applyFilterPushdownToScan` for it?

https://github.com/oap-project/gluten/blob/c531abd94045db71a8f8ef692e5c5a80cbcd118f/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L140-L145

Reply:
Fix this in https://github.com/oap-project/gluten/commit/d879fd097dada9aba9c21046b01b1104f916188a.